src/routes/v3/oauth_clients.rs

```diff
@@ -0,0 +1,387 @@
+use std::{collections::HashSet, fmt::Display};
+
+use actix_web::{
+    delete, get, patch, post,
+    web::{self},
+    HttpRequest, HttpResponse,
+};
+use chrono::Utc;
+use itertools::Itertools;
+use rand::{distributions::Alphanumeric, Rng, SeedableRng};
+use rand_chacha::ChaCha20Rng;
+use serde::{Deserialize, Serialize};
+use sqlx::PgPool;
+use validator::Validate;
+
+use super::ApiError;
+use crate::{
+    auth::get_user_from_headers,
+    database::{
+        models::{
+            generate_oauth_client_id, generate_oauth_redirect_id,
+            oauth_client_authorization_item::OAuthClientAuthorization,
+            oauth_client_item::{OAuthClient, OAuthRedirectUri},
+            DatabaseError, OAuthClientId, User,
+        },
+        redis::RedisPool,
+    },
+    models::{self, oauth_clients::OAuthClientCreationResult, pats::Scopes},
+    queue::session::AuthQueue,
+    routes::v2::project_creation::CreateError,
+    util::validate::validation_errors_to_string,
+};
+
+use crate::database::models::oauth_client_item::OAuthClient as DBOAuthClient;
+
+pub fn config(cfg: &mut web::ServiceConfig) {
+    cfg.service(oauth_client_create);
+    cfg.service(oauth_client_edit);
+    cfg.service(oauth_client_delete);
+    cfg.service(get_user_clients);
+    cfg.service(get_user_oauth_authorizations);
+    cfg.service(revoke_oauth_authorization);
+}
+
+#[get("user/{user_id}/oauth_apps")]
```

Comment:
this route should just be /oauth_apps IMO. it's more consistent with how we do PATs and sessions and notifs

Reply:
Sure, that makes sense. Do you think it should still take in {user_id} or should it just return the current user's clients based on the header, like pats?

Reply:
Depends on the anticipated use of it. If it was checking *other* user's apps (as a moderator, for instance) then {user_id} would make sense, but given that the code only accepts the user itself, I would say it doesnt need user_id